### PR TITLE
Add Network name utility

### LINKED ISSUE #253: Network name utility
### What feature should we add?

It would be convenient to retrieve the network name by using `w3.network`, which could just call the utility with  `w3.eth.chain_id`.

A `name_from_chain_id` function is needed that will return the full name of the network. (example: "Ethereum Mainnet")
A `short_name_from_chain_id` function would return the short name. (example: "eth")

List of networks in json: https://chainid.network/chains_mini.json

A script was written to parse the list json and create a list of enums: https://gist.github.com/reedsa/addafe9eb852db3267c0e90f1ce9867e
An adaptation of the script could make it easy to update the list chain ids and names in our code.

### PR DESCRIPTION
**Pull request by DavidRomanovizc**
### What was wrong?

Related to Issue #253
Closes #253 

### How was it fixed?
Created dictionaries `network_names` and `network_short_names` and the list networks that contains objects of the Network class with data about the network, including `chain_id`, `name`, `shortName`, and `symbol`. And implemented three functions: the first one extracts data from Network, the second one returns the Network name, and the third one returns the short network name. Additionally, I covered the code with parameterized tests.

### Todo:
- [x] Clean up commit history

- [x] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-utils/blob/master/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

**Comment by reedsa** [comment]
Hey @DavidRomanovizc nice work here and great to see the tests. Thinking about long term maintenance, I think we should just load and parse the JSON straight from https://chainid.network/chains_mini.json and then just loop over those to initialize each `Network` object.

If you have a chance to update how the initialization is done we'll get these merged! Thanks.

**Comment by DavidRomanovizc** [comment]
> Hey @DavidRomanovizc nice work here and great to see the tests. Thinking about long term maintenance, I think we should just load and parse the JSON straight from https://chainid.network/chains_mini.json and then just loop over those to initialize each `Network` object.
> 
> If you have a chance to update how the initialization is done we'll get these merged! Thanks.

Hello! I have updated the code. Now we load data from JSON, but our ChainId is not updating. If we have to load a new network from JSON and there is no ChainId present, it will raise a ValueError. Maybe we need update class ChainId

**Comment by reedsa** [comment]
Got it, that makes sense. I'll make sure to update eth-typings with the latest networks periodically.

**Comment by DavidRomanovizc** [comment]
> Got it, that makes sense. I'll make sure to update eth-typings with the latest networks periodically.

Also I've been thinking about the fact that we can get the `symbol` from the JSON, so we won't need the ChainId. Can we do that or we need use `ChainId` in this task?

File: eth_utils/network.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+network_short_names_by_id = {
+    network.chain_id: network.shortName for network in networks
+}
+
+
+def network_from_chain_id(chain_id: int) -> Network:
+    try:
+        return networks_by_id[chain_id]
+    except KeyError:
+        raise ValidationError(f"Chain_id is not recognized: {chain_id}")
```
Comment (pacrob) [inline eth_utils/network.py:65]: I think it should probably be "chain_id is not recognized", just to directly match the variable.

Here and below.

**Comment by reedsa** [comment]
@DavidRomanovizc that's a good thought, though the Network typings are better off if we keep `ChainId`. It shouldn't take much to maintain the list in eth-typing, where I'll be adding a script to update that list for us.

**Review by fselmo — APPROVED** [review]
lgtm!

**Comment by reedsa** [comment]
@DavidRomanovizc one more thank you! Appreciate the contribution.

**Pull request merged**